### DISCUSSION TITLE
exclude the generated assets from the routes

### DISCUSSION THREAD

**Discussion by Albert-Gao** [expo/router#271] [Ideas]
![image](https://user-images.githubusercontent.com/18282328/192896196-c960bc3f-cca0-4174-9e20-1586868cf647.png)

the generated folder is from relay compiler, which always create the codegen close to the route which uses GraphQL query.

the way next.js handles this is to add a property for the ignored folder name:

https://nextjs.org/docs/advanced-features/compiler#relay

![image](https://user-images.githubusercontent.com/18282328/192896466-9d507819-bfd6-4af5-a02c-a2eca582fd87.png)

Can we have something like this here?

I think, maybe we can make it an array, so we can have multiple paths being ignored rather than just ignoring the codegen folder.

**Comment by EvanBacon** [comment]
Based on the linked docs, it seems like this writes the generated files outside of the `pages` directory, then the Next.js server redirects the request somehow.

**Comment by jesuscovam** [comment]
maybe in the implementation of _getRecursiveTree_ or _contextModuleToFileNodes_ we could add a filter  for an excludePaths array, but we would also need to remove these files from the bundle

**Comment by MoSattler** [comment]
any news on this?

**Comment by oikalyptus** [comment]
This would be awesome. I like to put my graphql files close to where they are used.

**Comment by MoSattler** [comment]
any chance there are any updates?

**Comment by aflekkas** [comment]
Any updates? I'd love for this to become a feature... Maybe this should be re-discussed?
